pkg/distributor/errors.go

```diff
@@ -50,6 +61,12 @@ func (e replicasDidNotMatchError) Error() string {
 	return fmt.Sprintf("replicas did not match, rejecting sample: replica=%s, elected=%s", e.replica, e.elected)
 }
 
+func (e replicasDidNotMatchError) errorCause() mimirpb.ErrorCause {
+	return mimirpb.REPLICAS_DID_NOT_MATCH
+}
+
+var _ distributorError = replicasDidNotMatchError{}
```

Comment:
What are these statements for? is that to verify that the type satisfies the interface?
It would be useful (for me) if there was a comment what this does.

Reply:
Hi @replay, 
Exactly, these statements are here for verifying that the type satisfies the interface.
I was asked by @colega ([here](https://github.com/grafana/mimir/pull/6377#discussion_r1360422193)) to add them because this way the compiler complains if we by mistake forget to implement some methods of an interface.
I agree that adding a comment explaining that would be beneficial and will add it.

Reply:
I'm okay with having a comment, although this is a standard Golang statement.